### PR TITLE
Replace once_cell::Lazy with std::sync::OnceLock for global Initialization in otel-sdk crate

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

changes similar to #2326 for otel-sdk.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/trace/provider.rs [hunk 1]
```diff
@@ -68,32 +68,39 @@ use crate::trace::{
...
 use opentelemetry::{otel_debug, trace::TraceResult};
 use std::borrow::Cow;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::Arc;
+use std::sync::{Arc, OnceLock};
 
 use super::IdGenerator;
 
-static PROVIDER_RESOURCE: OnceCell<Resource> = OnceCell::new();
+// TODO Replace with LazyLock once it is stable
```
Comment (utpilla) [inline opentelemetry-sdk/src/trace/provider.rs:80]: nit: We might not be able to use `LazyLock` for `PROVIDER_RESOURCE` as `LazyLock` doesn’t allow additional inputs to its function after you call [LazyLock::new(|| ...)](https://doc.rust-lang.org/stable/std/sync/struct.LazyLock.html#method.new).

**Review by utpilla — APPROVED** [review]

File: opentelemetry-sdk/src/trace/provider.rs (hunk 1 above)
Comment (lalitb) [reply opentelemetry-sdk/src/trace/provider.rs:80]: good catch :)

**Comment by lalitb** [comment]
The CI build is failing unrelated to this CI:

```
  Compiling sysinfo v0.30.13
error[E0599]: no method named `len` found for type `i8` in the current scope
   --> /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sysinfo-0.30.13/src/unix/apple/macos/process.rs:436:53
    |
436 |         Some(node.vip_path.len() * node.vip_path[0].len()),
    |                                                     ^^^
    |
help: there is a method `le` with a similar name, but with different arguments
   --> /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14/library/core/src/cmp.rs:1196:5

For more information about this error, try `rustc --explain E0599`.
error: could not compile `sysinfo` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

Have raised a bug for sysinfo for  now - https://github.com/GuillaumeGomez/sysinfo/issues/1392

**Review by hdost — APPROVED** [review]
One less dependency 🤘

**Pull request merged**